_pages/algorithm-development/languages/python.md

```diff
@@ -10,57 +10,90 @@ image:
+When executed in an Algorithmia container and providing the string "HAL 9000" as an input, this algorithm will output "Hello HAL 9000". If executed locally, the algorithm will print `Hello Algorithmia` to `stdout` instead.
 
-<img src="{{site.cdnurl}}{{site.baseurl}}/images/post_images/algo_dev_lang/algorithm_console_python.png" alt="Algorithm console Python" class="screenshot">
+## Loaded State
 
+Developing with the ADK allows you to make use of an optional `load()` function for preparing an algorithm for runtime operations. Using `load()`, we can load state into memory prior to a function's execution and then pass that data to `apply()` by adding an additional `globals` parameter in the `apply` function. 
 
-## Managing Dependencies
+Let's add a `load()` function to the Hello World example we just created:
 
-Algorithmia supports adding 3rd party dependencies via the <a href="https://pypi.python.org/pypi">Python Package Index (PyPI)</a> using a requirements.txt file.
+{% highlight python %}
+from Algorithmia import ADK
 
-On the algorithm editor page there is a button on the top right that says "Dependencies". Click that button and you'll see a modal window:
+def apply(input, globals):
+    return "hello {} {}".format(str(input), str(globals['payload']))
 
-<img src="{{site.cdnurl}}{{site.baseurl}}/images/post_images/algo_dev_lang/dependencies_python.png" alt="Python Dependency File" class="screenshot img-md">
+def load():
+    globals = {}
+    globals['payload'] = "Loading has been completed."
+    return globals
 
-If you have any dependencies you can add them by typing in the package name to the `requirements.txt` file.
+algorithm = ADK(apply, load)
+algorithm.init("Algorithmia")
+{% endhighlight %}
 
-This guide won't depend on any external dependencies so you can close the dependencies window.
+In our load function we've created a `globals` object and added a key called `payloads` with a string value `Loading has been completed.`; we then return the globals object, which will be passed as input to the algorithm's `apply` function, where we use that value as part of the algorithm's output. Executing the algorithm locally will result in `Hello Algorithmia Loading has been completed.` being printed to `stdout`.
 
-If you do add dependencies, you will still need to import those packages via the import statement to your algorithm file as you would do for any Python script.
-{: .notice-info}
+## Available Libraries
+
+In addition to your own code in `src/Algorithm.py`, Algorithmia makes a number of libraries available to make algorithm development easier. We support multiple versions of Python and a variety of frameworks, and we continue to add new variants and broaden GPU support. A complete list of supported environments can be found on the [Enviornment Matrix](/developers/model-deployment/environments/) page, and are available through the "Environment" drop-down when creating a new algorithm.
```

Comment:
correct typo "Enviornment" ;
change "/model-deployment/" to "/algorithm-development/" in link ; this page just moved as of last week :)

Reply:
"supported environments" --> "predefined environments" (don't want anyone to think we don't support other libraries)